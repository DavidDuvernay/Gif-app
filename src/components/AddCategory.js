import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories, categories } ) => {
 
    const [inputValue, setInputValue] = useState('');

    const handleinputChange = ( e ) => {
        
        setInputValue( e.target.value )
    };

    const handleSubmit =( e ) => {
        e.preventDefault();

        setCategories(categories => [ ...categories, inputValue]);
        setInputValue('');
    };

    return (
        <form onSubmit = { handleSubmit }>

            <input
                type = "text"
                value = { inputValue }
                onChange = { handleinputChange }
            /> 

        </form>
    )
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};