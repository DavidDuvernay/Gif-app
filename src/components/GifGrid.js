import React from 'react';

export const GifGrid = ( { category }) => {
    
    const getGifs = async() => {
        const url = 'http://api.giphy.com/v1/gifs/search?q=Rick%20and%20Morty&limit=10&api_key=DjBrCGeFD2LhaQNB1Lb05osHuPubFNcg';
        
        const resp = await fetch( url );
        
        const { data } = await resp.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });

        console.log(gifs)
    };

    getGifs()

    return(
        <div> 
            <h3> { category } </h3>
        </div>
    );
    
}