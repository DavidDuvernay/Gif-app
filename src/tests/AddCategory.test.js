import React from 'react';
import { AddCategory } from '../components/AddCategory';
const { shallow } = require("enzyme");
import '@testing-library/jest-dom';

describe('Pruebas en el componente', () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( < AddCategory setCategories = { setCategories }/> );
    })
    
    test('Debe de mostrar el componente correctamente', () => {
    
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('Debe simular cambio en la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola Mundo'
        input.simulate('change', { target: { value: value } });

        expect( wrapper.find('p').text().trim() ).toBe( value );

    });

    test('No debe de postear la informacion onSubmit', () => {

        wrapper.find('form').simulate('submit', { preventDefault: () => {} });

        expect( setCategories).not.toHaveBeenCalled();
    });

    test('Debe de llamar setCategories y limpiar el campo de texto', () => {

        const value = 'Hola Mundo';

        wrapper.find('input').simulate('change', { target: { value } });

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( wrapper.find('input').prop('value')).toBe('');
        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories).toHaveBeenCalledTimes(1);
        expect( setCategories).toHaveBeenCalledWith( expect.any( Function) );
    });
});