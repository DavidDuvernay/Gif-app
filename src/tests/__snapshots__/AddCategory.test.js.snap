// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Debe de mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
